Mathlib/Probability/Distributions/Gaussian.lean

```diff
@@ -0,0 +1,194 @@
+      mul_inv_rev, NNReal.coe_mul, NNReal.coe_inv, NNReal.coe_ofNat, neg_mul, mul_eq_mul_left_iff,
+      Real.exp_eq_exp, mul_eq_zero, inv_eq_zero, Real.sqrt_eq_zero, NNReal.coe_eq_zero, hv,
+      false_or]
+    rw [mul_comm]
+    left
+    field_simp
+  exact Integrable.comp_sub_right hg μ
+
+/-- The gaussian distribution pdf integrates to 1 when the variance is not zero.  -/
+lemma lintegral_gaussianPdfReal_eq_one (μ : ℝ) {v : ℝ≥0} (h : v ≠ 0) :
+    ∫⁻ x, ENNReal.ofReal (gaussianPdfReal μ v x) = 1 := by
+  rw [←ENNReal.toReal_eq_one_iff]
+  have hfm : AEStronglyMeasurable (gaussianPdfReal μ v) ℙ :=
+    (stronglyMeasurable_gaussianPdfReal μ v).aestronglyMeasurable
+  have hf : 0 ≤ₐₛ gaussianPdfReal μ v := ae_of_all _ (gaussianPdfReal_nonneg μ v)
+  rw [← integral_eq_lintegral_of_nonneg_ae hf hfm]
+  simp only [gaussianPdfReal, gt_iff_lt, zero_lt_two, zero_le_mul_right, ge_iff_le, one_div,
+    Nat.cast_ofNat, integral_mul_left]
+  rw [integral_sub_right_eq_self (μ := ℙ) (fun a ↦ rexp (-a ^ 2 / ((2 : ℝ) * v))) μ]
+  simp only [gt_iff_lt, zero_lt_two, zero_le_mul_right, ge_iff_le, div_eq_inv_mul, mul_inv_rev,
+    mul_neg]
+  simp_rw [←neg_mul]
+  rw [neg_mul, integral_gaussian, ← Real.sqrt_inv, ←Real.sqrt_mul]
+  · field_simp
+    ring
+  · positivity
+
+/-- The gaussian distribution pdf integrates to 1 when the variance is not zero.  -/
+lemma integral_gaussianPdfReal_eq_one (μ : ℝ) {v : ℝ≥0} (hv : v ≠ 0) :
+    ∫ x, gaussianPdfReal μ v x = 1 := by
+  have h := lintegral_gaussianPdfReal_eq_one μ hv
+  rw [← ofReal_integral_eq_lintegral_ofReal (integrable_gaussianPdfReal _ _)
+    (ae_of_all _ (gaussianPdfReal_nonneg _ _)), ← ENNReal.ofReal_one] at h
+  rwa [← ENNReal.ofReal_eq_ofReal_iff (integral_nonneg (gaussianPdfReal_nonneg _ _)) zero_le_one]
+
+/-- The pdf of a Gaussian distribution on ℝ with mean `μ` and variance `v`. -/
+noncomputable
+def gaussianPdf (μ : ℝ) (v : ℝ≥0) (x : ℝ) : ℝ≥0∞ := ENNReal.ofReal (gaussianPdfReal μ v x)
+
+lemma gaussianPdf_def (μ : ℝ) (v : ℝ≥0) :
+    gaussianPdf μ v = fun x ↦ ENNReal.ofReal (gaussianPdfReal μ v x) := rfl
+
+@[simp]
+lemma gaussianPdf_zero_var (μ : ℝ) : gaussianPdf μ 0 = 0 := by
+  ext
+  simp [gaussianPdf]
+
+lemma gaussianPdf_pos (μ : ℝ) {v : ℝ≥0} (hv : v ≠ 0) (x : ℝ) : 0 < gaussianPdf μ v x := by
+  rw [gaussianPdf, ENNReal.ofReal_pos]
+  exact gaussianPdfReal_pos _ _ _ hv
+
+@[measurability]
+lemma measurable_gaussianPdf (μ : ℝ) (v : ℝ≥0) : Measurable (gaussianPdf μ v) :=
+  (measurable_gaussianPdfReal _ _).ennreal_ofReal
+
+@[simp]
+lemma lintegral_gaussianPdf_eq_one (μ : ℝ) {v : ℝ≥0} (h : v ≠ 0) :
+    ∫⁻ x, gaussianPdf μ v x = 1 :=
+  lintegral_gaussianPdfReal_eq_one μ h
+
+end GaussianPdf
+
+section GaussianReal
+
+/-- A Gaussian distribution on `ℝ` with mean `μ` and variance `v`. -/
+noncomputable
+def gaussianReal (μ : ℝ) (v : ℝ≥0) : Measure ℝ :=
+  if v = 0 then Measure.dirac μ else volume.withDensity (gaussianPdf μ v)
+
+@[simp]
```

Comment:
do you really want this to be a simp lemma? I'd be happy to hide things behind the definition, not to be broken out by simp. If you need it for building your API, it might make sense to use `attribute [local simp] gaussianReal_of_var_ne_zero`.